### PR TITLE
Improve JSDoc such that `ng build --prod` with angular/cli 6.0.8 succeeds again

### LINKED ISSUE #8421: JSDoc error when compiling with Angular CLI 6.0.8
When trying to compile a project using `ng build --prod` with Angular CLI 6.0.8 and OpenLayers 5.0.0 and the project includes for example Feature.js using `import Feature from 'ol/Feature';`, the following error occurs:

`Module build failed: TypeError: Cannot read property 'kind' of undefined`

I've debugged the error back to the line:

` * @param {(module:ol/format/Feature~WriteOptions|module:ol/format/Feature~ReadOptions)=} opt_options`

When changing this line to

` * @param {module:ol/format/Feature~WriteOptions|module:ol/format/Feature~ReadOptions=} opt_options`

the error doesn't occur.

### PR DESCRIPTION
**Pull request by jkoelewijn**
Fixes #8421.

### CODE REVIEW

File: src/ol/format/Feature.js [hunk 1]
```diff
@@ -200,7 +200,7 @@ export default FeatureFormat;
 /**
  * @param {module:ol/geom/Geometry|module:ol/extent~Extent} geometry Geometry.
  * @param {boolean} write Set to true for writing, false for reading.
- * @param {(module:ol/format/Feature~WriteOptions|module:ol/format/Feature~ReadOptions)=} opt_options
+ * @param {module:ol/format/Feature~WriteOptions|module:ol/format/Feature~ReadOptions=} opt_options
```
Comment (ahocevar) [inline src/ol/format/Feature.js]: This one should probably become
```js
 * @param {(module:ol/format/Feature~WriteOptions=|module:ol/format/Feature~ReadOptions=} opt_options
```

File: src/ol/PluggableMap.js [hunk 2]
```diff
@@ -61,7 +61,7 @@ import {create as createTransform, apply as applyTransform} from './transform.js
 
 /**
  * @typedef {Object} AtPixelOptions
- * @property {((function(module:ol/layer/Layer): boolean)|undefined)} layerFilter Layer filter
+ * @property {function(module:ol/layer/Layer): boolean|undefined} layerFilter Layer filter
```
Comment (tschaub) [inline src/ol/PluggableMap.js]: This makes the type ambiguous.  The type could be a function or undefined.  As it is written here, the function could return a boolean or undefined.

Comment (ahocevar) [reply src/ol/PluggableMap.js]: Right. In this case you‘d have to put `undefined` first, and then the function.

**Comment by jkoelewijn** [comment]
Thanks for your comments. I've added some commits to improve the pull request.

**Comment by tschaub** [comment]
Thanks @jkoelewijn.  We're working toward using TypeScript for type checking (see #8345).  I'm hoping that is what the Angular CLI uses too.  Otherwise, I think it will be hard to maintain a type syntax that doesn't break the `ng` tool.

**Pull request merged**